test/chain-test.js

```diff
@@ -730,6 +731,289 @@ describe('Chain', function() {
     });
   });
 
+  describe('Covenant limits', function() {
+    before(async () => {
+      await chain.reset(0);
+      wallet = new MemWallet({ network });
```

Comment:
nit, You could use `const` here, no need to reuse the same var.

Reply:
Actually I can't, the reference to `wallet` is defined globally all the way up on line 49! And `wallet` is used throughout the test. What I should do is redefine a local `let wallet` inside this `describe` block, will do.

Reply:
probably redefine new chain as well?

Reply:
okie doke, one sec...